numba/cuda/codegen.py

```diff
@@ -134,40 +137,56 @@ def _get_ptxes(self, cc=None):
 
         irs = self.llvm_strs
 
-        ptxes = [nvvm.llvm_to_ptx(irs, **options)]
+        ptx = nvvm.compile_ir(irs, **options)
 
         # Sometimes the result from NVVM contains trailing whitespace and
         # nulls, which we strip so that the assembly dump looks a little
         # tidier.
-        ptxes = [x.decode().strip('\x00').strip() for x in ptxes]
+        ptx = ptx.decode().strip('\x00').strip()
 
         if config.DUMP_ASSEMBLY:
             print(("ASSEMBLY %s" % self._name).center(80, '-'))
-            print(self._join_ptxes(ptxes))
+            print(ptx)
             print('=' * 80)
 
-        self._ptx_cache[cc] = ptxes
+        self._ptx_cache[cc] = ptx
+
+        return ptx
+
+    def get_ltoir(self, cc=None):
+        cc = self._ensure_cc(cc)
+
+        ltoir = self._ltoir_cache.get(cc, None)
+        if ltoir:
+            return ltoir
 
-        return ptxes
+        arch = nvvm.get_arch_option(*cc)
+        options = self._nvvm_options.copy()
+        options['arch'] = arch
+        options['gen-lto'] = None
 
-    def _join_ptxes(self, ptxes):
-        return "\n\n".join(ptxes)
+        irs = self.llvm_strs
+        ltoir = nvvm.compile_ir(irs, **options)
+        self._ltoir_cache[cc] = ltoir
+
+        return ltoir
 
     def get_cubin(self, cc=None):
-        if cc is None:
-            ctx = devices.get_context()
-            device = ctx.device
-            cc = device.compute_capability
+        cc = self._ensure_cc(cc)
 
         cubin = self._cubin_cache.get(cc, None)
         if cubin:
             return cubin
 
         linker = driver.Linker.new(max_registers=self._max_registers, cc=cc)
 
-        ptxes = self._get_ptxes(cc=cc)
-        for ptx in ptxes:
+        if linker.lto:
+            ltoir = self.get_ltoir(cc=cc)
+            linker.add_ltoir(ltoir)
```

Comment:
This is untested as a dependency on `pynvjitlink` would be needed to do such a test, on inspection only it seems ok.